src/plugins/discover/public/context_awareness/profile_providers/observability/observability_root_profile/accessors/get_app_menu.ts

```diff
@@ -0,0 +1,154 @@
+
+    return {
+      appMenuRegistry: (registry) => {
+        // Register custom link actions
+        registerDatasetQualityLink(registry, services);
+        // Register alerts sub menu actions
+        registerCreateSLOAction(registry, services, params);
+        registerCustomThresholdRuleAction(registry, services, params);
+
+        return prevValue.appMenuRegistry(registry);
+      },
+    };
+  };
+
+const registerDatasetQualityLink = (
+  registry: AppMenuRegistry,
+  { share, timefilter }: ProfileProviderServices
+) => {
+  const dataQualityLocator =
+    share?.url.locators.get<DataQualityLocatorParams>(DATA_QUALITY_LOCATOR_ID);
+
+  if (dataQualityLocator) {
+    registry.registerCustomAction({
+      id: 'dataset-quality-link',
+      type: AppMenuActionType.custom,
+      controlProps: {
+        label: 'Data sets',
+        testId: 'dataset-quality-link',
+        onClick: ({ onFinishAction }) => {
+          const refresh = timefilter.getRefreshInterval();
+          const { from, to } = timefilter.getTime();
+
+          dataQualityLocator.navigate({
+            filters: {
+              timeRange: {
+                from: from ?? 'now-24h',
+                to: to ?? 'now',
+                refresh,
+              },
+            },
+          });
+
+          onFinishAction();
+        },
+      },
+    });
+  }
+};
+
+const registerCustomThresholdRuleAction = (
+  registry: AppMenuRegistry,
+  { data, triggersActionsUi }: ProfileProviderServices,
+  { dataView }: AppMenuExtensionParams
+) => {
+  registry.registerCustomActionUnderSubmenu(AppMenuActionId.alerts, {
+    id: AppMenuActionId.createRule,
+    type: AppMenuActionType.custom,
+    order: 101,
+    controlProps: {
+      label: 'Create custom threshold rule',
+      iconType: 'visGauge',
+      testId: 'custom-threshold-rule',
+      onClick: ({ onFinishAction }) => {
+        const index = dataView?.getIndexPattern();
+        const { filters, query } = data.query.getState();
+
+        return triggersActionsUi.getAddRuleFlyout({
+          consumer: 'logs',
+          ruleTypeId: OBSERVABILITY_THRESHOLD_RULE_TYPE_ID,
+          canChangeTrigger: false,
+          initialValues: {
+            params: {
+              searchConfiguration: {
+                index,
+                query,
+                filter: filters,
+              },
+            },
+          },
+          onClose: onFinishAction,
+        });
+      },
+    },
+  });
+};
+
+const registerCreateSLOAction = (
+  registry: AppMenuRegistry,
+  { data, discoverShared }: ProfileProviderServices,
+  { dataView, isEsqlMode }: AppMenuExtensionParams
+) => {
+  const sloFeature = discoverShared.features.registry.getById('observability-create-slo');
+
+  if (sloFeature) {
+    registry.registerCustomActionUnderSubmenu(AppMenuActionId.alerts, {
+      id: 'create-slo',
+      type: AppMenuActionType.custom,
+      order: 102,
+      controlProps: {
+        label: 'Create SLO',
```

Comment:
And one last spot to use i18n 🙂

Reply:
Addressed in [refactor(discover): use i18n](https://github.com/elastic/kibana/pull/198987/commits/8c073e23cff9d1566718a93645c8a522cf2e9980)